pyccel/codegen/printing/cwrappercode.py

```diff
@@ -153,7 +153,7 @@ def _get_static_function(self, used_names, function, collect_dict):
                         var = Variable(dtype=NativeInteger() ,name = self.get_new_name(used_names, a.name + "_dim"))
                         body = FunctionCall(numpy_get_dim, [collect_dict[a], i])
                         if a.is_optional:
-                            body = IfTernaryOperator(VariableAddress(collect_dict[a]), body , LiteralInteger(0))
+                            body = IfTernaryOperator(PyccelIsNot(VariableAddress(collect_dict[a]),Nil()), body , LiteralInteger(0))
```

Comment:
I this a bug fix? If yes, please add a line about it in the PR's description

Reply:
Yes and no. Previously the output was:
```
a = b ?  c : d;
```
whereas now it is:
```
a = b == NULL ?  c : d;
```
The second option is clearer and allows me to find and exclude this case rather than trying to unravel it as `b` is an array

Reply:
I have added a line